src/main/java/mekanism/common/content/gear/mekatool/ModuleVeinMiningUnit.java

```diff
@@ -58,8 +59,8 @@ public static Map<BlockPos, BlockState> findPositions(Level world, Map<BlockPos,
         openSet.putAll(initial);
         int singleMaxCount = MekanismConfig.gear.disassemblerMiningCount.get() - 1;
         int maxCount = singleMaxCount * initial.size();
-        int next = openSet.size();
-        int dist = 0;
+        AtomicInteger next = new AtomicInteger(openSet.size());
+        AtomicInteger dist = new AtomicInteger(0);
```

Comment:
What exactly is the scope issue here? Is it just a warning or is there actually some issue

Reply:
Modifying variables from within a while loop shouldn't have any issues in relation to their values reverting? Unless you are declaring them inside the loop of course.

Reply:
for loop did work, although I think it still has the warning

Reply:
I will look at it a bit when I am testing this PR then and see if I can figure out why it has issues when it isn't atomic.

Reply:
Java does some of these things weird, I think it's basically doing a separate stack frame for in the loop and the variables outside are treated more like params. I've had issues similar to this before. It doesn't have to be atomic as it's not multi threaded, it would work with any object with an int field to get/set that was just what the ide wanted to replace it with